packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -1034,42 +1383,48 @@ impl<'m> RunnerImpl<'m> {
         // undefined behavior, because the pointer to the schema comes from
         // an immutable reference.
         // ---> Do *not* mutate the schema bytes in `runner.js`.
-        let mut agent_schema_bytes =
-            schema_to_stream_bytes(&run.datastore.agent_batch_schema.arrow);
-        let mut msg_schema_bytes = schema_to_stream_bytes(&run.datastore.message_batch_schema);
-        let mut ctx_schema_bytes = schema_to_stream_bytes(&run.datastore.context_batch_schema);
+        let agent_schema_bytes = schema_to_stream_bytes(&run.datastore.agent_batch_schema.arrow);
+        let msg_schema_bytes = schema_to_stream_bytes(&run.datastore.message_batch_schema);
+        let ctx_schema_bytes = schema_to_stream_bytes(&run.datastore.context_batch_schema);
         // run.shared_context.datasets?
 
         // Keep schema vecs alive while bytes are passed to V8.
-        let agent_schema_bytes = bytes_to_js(mv8, &mut agent_schema_bytes);
-        let msg_schema_bytes = bytes_to_js(mv8, &mut msg_schema_bytes);
-        let ctx_schema_bytes = bytes_to_js(mv8, &mut ctx_schema_bytes);
+        let agent_schema_bytes = bytes_to_js(scope, &agent_schema_bytes);
+        let msg_schema_bytes = bytes_to_js(scope, &msg_schema_bytes);
+        let ctx_schema_bytes = bytes_to_js(scope, &ctx_schema_bytes);
 
-        let pkg_ids = mv8.create_array();
-        let pkg_msgs = mv8.create_array();
+        let pkg_ids = v8::Array::new(scope, run.packages.0.len() as i32);
+        let pkg_msgs = v8::Array::new(scope, run.packages.0.len() as i32);
         for (i_pkg, (pkg_id, pkg_msg)) in run.packages.0.iter().enumerate() {
             let i_pkg = i_pkg as u32;
-            pkg_ids.set(i_pkg, pkg_id_to_js(mv8, *pkg_id))?;
+            let js_pkg_id = pkg_id_to_js(scope, *pkg_id);
+            pkg_ids
+                .set_index(scope, i_pkg, js_pkg_id)
+                .ok_or_else(|| format!("Could not set pkg_ids at index {i_pkg}"))?;
```

Comment:
Same comment as all of the ones above about could not set index

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/2a832f8dc26357b06320aed8e66cb59822d849a2